packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,333 @@
+# Mapping Table Column
+
+## Overview
+
+The `nimble-table-column-mapping` is a component that supports rendering specific number, boolean, or string values as mapped, predefined, html fragments (with content restrictions).
+
+### Background
+
+[Icon column type issue](https://github.com/ni/nimble/issues/1013)
+
+[Boolean column type issue](https://github.com/ni/nimble/issues/1103)
+
+### Features
+
+-   Supported input:
+    -   string
+    -   number
+    -   boolean
+-   Supported output:
+    -   Text
+    -   One or more Nimble icons (with colors via icon severity property)
+    -   Nimble spinner
+    -   One span of text preceded and/or followed by icons/spinner
+    -   (empty)
+
+### Non-goals
+
+-   Non-Nimble icon support
+-   Arbitrary icon colors
+-   Hyperlink icons
+-   Non-icon, non-spinner Nimble components
+-   Arbitrary html
+
+---
+
+## Design
+
+In addition to the `nimble-table-column-mapping` element, there are `nimble-mapping-<input type>` elements to define mappings between values and html fragments. Below is an example of how these elements would be used within a `nimble-table`:
```

Comment:
It's not clear to me the benefit of having multiple mapping elements based on input type.

A column can't mix types, it's either number|null, string|null, boolean|null, etc. So it should be defined per column, not per mapping entry.

So why define the elements with the flexibility that they can mix multiple representations within a column? Just makes for extra validation work.

Could `nimble-table-column-mapping` have a type property and then have a single mapping element of `key: type boolean | string | number`?

Reply:
Taking suggested approach